src/DynamoCore/Models/DynamoModelCommands.cs

```diff
@@ -66,35 +66,56 @@ void CreateNodeImpl(CreateNodeCommand command)
                     var name = command.Name;
                     var nodeId = command.NodeId;
 
-                    Guid customNodeId; // To be used in the event it's a custom node we're making.
+                    // find nodes with of the same type with the same GUID
+                    var query = CurrentWorkspace.Nodes.Where(n => n.GUID.Equals(nodeId) && n.Name.Equals(name));
 
-                    // Then, we have to figure out what kind of node to make, based on the name.
-
-                    // First, we check for a DSFunction by looking for a FunctionDescriptor
-                    var functionItem = LibraryServices.GetFunctionDescriptor(name);
-                    if (functionItem != null)
-                    {
-                        node = (functionItem.IsVarArg)
-                            ? new DSVarArgFunction(functionItem) as NodeModel
-                            : new DSFunction(functionItem);
-                        node.GUID = nodeId;
-                    }
-                    // If that didn't work, let's try using the NodeFactory
-                    else if (NodeFactory.CreateNodeFromTypeName(name, out node))
-                    {
-                        node.GUID = nodeId;
-                    }
-                    // And if that didn't work, then it must be a custom node.
-                    else if (Guid.TryParse(name, out customNodeId))
+                    // safely ignore a node of the same type with the same GUID
+                    if (query.Any())
                     {
-                        node = CustomNodeManager.CreateCustomNodeInstance(customNodeId);
-                        node.GUID = nodeId;
+                        node = query.First();
                     }
```

Comment:
we can't return from here because there are some code after these if-else statements